config/cylc_hpc/flow.cylc

```diff
@@ -106,82 +102,89 @@
+            singularity exec --bind $results_dir/${location_array[$i]}/landmasks:/tmp,$report_dir:/usr/local/bin/../report $project_dir/icefloetracker-julia.simg $julia_exec -t auto /usr/local/bin/ice-floe-tracker.jl landmask $fetchdata_dir/${location_array[$i]} /tmp
         """
         platform = oscar
         execution time limit = PT1H
         submission retry delays = 2*PT30S
-        execution retry delays = 2*PT6S
+        execution retry delays = PT6S
         [[[directives]]]
             --mem = 20G
             --cpus-per-task = 16
     
-    [[preprocess<job_id>]]
+    [[preprocess<param_set>]]
         script = """
-            i=${CYLC_TASK_PARAM_job_id}
+            i=${CYLC_TASK_PARAM_param_set}
+            location_array=( $(echo $location | sed -e 's/,/ /g') )
 
-            singularity exec --bind $preprocess_dir/$i:/tmp,$report_dir:/usr/local/bin/../report $project_dir/icefloetracker-julia.simg $julia_exec -t auto /usr/local/bin/ice-floe-tracker.jl preprocess -t $fetchdata_dir/$i/truecolor -r $fetchdata_dir/$i/reflectance -l $landmask_dir/$i -p $soit_dir/$i -o /tmp
+            singularity exec --bind $results_dir/${location_array[$i]}/preprocess:/tmp,$report_dir:/usr/local/bin/../report $project_dir/icefloetracker-julia.simg $julia_exec -t auto /usr/local/bin/ice-floe-tracker.jl preprocess -t $fetchdata_dir/${location_array[$i]}/truecolor -r $fetchdata_dir/${location_array[$i]}/reflectance -l $results_dir/${location_array[$i]}/landmasks -p $results_dir/${location_array[$i]}/soit -o /tmp
         """
         platform = oscar
         execution time limit = PT1H
         submission retry delays = 2*PT30S
-        execution retry delays = 2*PT6S
+        execution retry delays = PT6S
         [[[directives]]]
             --mem = 60G
             --cpus-per-task = 40
     
-    [[extractfeatures<job_id>]]
+    [[extractfeatures<param_set>]]
         script = """
-            i=${CYLC_TASK_PARAM_job_id}
+            i=${CYLC_TASK_PARAM_param_set}
+            location_array=( $(echo $location | sed -e 's/,/ /g') )
+            preprocess_dir=$results_dir/${location_array[$i]}/preprocess
 
-            singularity exec --bind $preprocess_dir/$i:/tmp,$report_dir:/usr/local/bin/../report $project_dir/icefloetracker-julia.simg $julia_exec -t auto /usr/local/bin/ice-floe-tracker.jl extractfeatures -i $preprocess_dir/$i -o /tmp --minarea $minfloearea --maxarea $maxfloearea
+            singularity exec --bind $preprocess_dir:/tmp,$report_dir:/usr/local/bin/../report $project_dir/icefloetracker-julia.simg $julia_exec -t auto /usr/local/bin/ice-floe-tracker.jl extractfeatures -i $preprocess_dir -o /tmp --minarea $minfloearea --maxarea $maxfloearea
         """
         platform = oscar
         execution time limit = PT1H
         submission retry delays = 2*PT30S
-        execution retry delays = 2*PT6S
+        execution retry delays = PT6S
         [[[directives]]]
             --mem = 20G
             --cpus-per-task = 8
```

Comment:
Same as number of images? Although this doesn't take long either. Could maybe be just 1.